src/Context/QuizProvider.tsx

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useReducer } from "react";
+import { initialState, quizReducer } from "../Reducer/quizReducer";
+import { ContextType } from "./Context.type";
+
+const QuizContext = createContext<ContextType>({} as ContextType);
+
+export const useQuizContext = () => useContext(QuizContext);
+
+export const QuizProvider: React.FC = ({ children }) => {
+  const [state, dispatch] = useReducer(quizReducer, initialState);
```

Comment:
I would suggest having specific names for state and dispatch, you can name it quizState and quizDispatch, because later when the code grows, you might need another context and another reducer (auth for example) for which you would need to name specifically